front/src/modules/simulationResult/components/SpaceTimeChartWrapper/SpaceTimeChartWrapper.tsx

```diff
@@ -211,10 +211,40 @@ const SpaceTimeChartWrapper = ({
     allTrainsProjected
   );
 
-  const splitPoints = useMemo<SplitPoint[]>(() => {
-    const pathsIndex = keyBy(paths, ({ id }) => id);
+  const pathsById = useMemo(() => keyBy(paths, ({ id }) => id), [paths]);
 
-    return (
+  const toPaced = (id: string) => {
+    if (isOccurrenceId(id)) return extractPacedTrainIdFromOccurrenceId(id);
+    if (isPacedTrainId(id)) return id;
+    return undefined;
+  };
+
+  const countByPacedAndTrack = (zones?: OccupancyZone[]) => {
```

Comment:
You can remove the `?? []` l.224

```suggestion
  const countByPacedAndTrack = (zones: OccupancyZone[] = []) => {
```